### ISSUE TITLE
[Catalog Import] Github integration: Support only imports from Entity Files (optionally turn off PR functionality)

### ISSUE THREAD

**Issue by geototti21** [backstage/backstage#4283]
## Feature Suggestion

Hey, we have a use case where we want to implement the [catalog-import](https://github.com/backstage/backstage/tree/master/plugins/catalog-import) plugin with GitHub integration but support only the option to register be providing the full URL to that Entity File and not provide the option to make a PR.

I can't find a way to turn this off, all the other integrations options work fine but look like [this specific check](https://github.com/backstage/backstage/blob/master/plugins/catalog-import/src/components/ImportComponentPage.tsx#L108) for GitHub doesn't allow it.

Let me know if I miss something here 😄 

Thanks team

## Possible Implementation

Extend the config with some kind of flag?

**Comment by freben** [comment]
I think this may be better suited as a prop to the plugin, than as a config option. To be added [here](https://github.com/backstage/backstage/blob/master/packages/app/src/App.tsx#L74), then.

I haven't heard of this requirement before now. Is it something you might be interested in contributing?

**Comment by dhenneke** [comment]
I'm actively working on the plugin in #4271 and also plan to add configurability for these features. I'm not sure if it will be part of the PR, but I anticipate to at least make it easy to add it in follow-ups.

**Comment by geototti21** [comment]
Thanks,

@freben is there any example of how this can be achieved via prop? Should we provide an integration object, eg:

_forgive my pseudocode_

```js
{
 "github":
    "open_pr": false,
    "import_from_file: true
}
```

And if something is not in this prop object it will default to `true` with a combination of the `integration` config.


@dhenneke your PR looks great, can somehow make the change for my suggestion compatible with your work in advance or we should keep it separate? :)

**Comment by geototti21** [comment]
👋 hey @freben , I made a quick Draft PR to have more data to discuss https://github.com/backstage/backstage/pull/4331

let me know what you think 😃

**Comment by geototti21** [comment]
feature was implemented to #4271
